packages/server/validators/users.js

```diff
@@ -50,7 +51,113 @@ const userIdBodyValidator = [
+        if (user.isNew === false) {
+          throw new Error("User already onboarded!");
+        }
+      } catch (error) {
+        if (error.message === "User already onboarded!") {
+          throw new Error("User already onboarded!");
+        } else {
+          throw new Error("User not found!");
+        }
+      }
+    }),
+
+  // First Name Check: exist, be between 1 and 50 chars
+  body("firstName", "Invalid firstName.")
+    .exists({ checkFalsy: true })
+    .withMessage("firstName can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 50 })
+    .withMessage("firstName length must be between 1 to 255 characters.")
+    .isAlpha()
+    .withMessage("firstName should be Alphabetic characters."),
+
+  // Last Name Check: exist, be between 1 and 50 chars
+  body("lastName", "Invalid lastName.")
+    .exists({ checkFalsy: true })
+    .withMessage("firstName can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 50 })
+    .withMessage("lastName length must be between 1 to 50 characters."),
+
+  // Pronoun Check: exist, be between 1 and 25 chars, and inside the allowed array.
+  body("pronouns", "Invalid pronoun.")
+    .exists({ checkFalsy: true })
+    .withMessage("pronoun can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 25 })
+    .withMessage("pronoun length must be between 1 to 25 characters.")
+    .isIn(ALLOWED_PRONOUN)
+    .withMessage(
+      "pronouns must be one of the allowed values: " +
+        ALLOWED_PRONOUN.join(", "),
+    ),
+
+  // Major Check: exist, be between 1 and 50 chars
+  body("major", "Invalid major.")
+    .exists({ checkFalsy: true })
+    .withMessage("major can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 25 })
+    .withMessage("Major length must be between 1 to 25 characters."),
+
+  // Avatar Check: exist, be between 1 and 255 chars, and is URL!
+  body("avatar", "Invalid avatar.")
+    .exists({ checkFalsy: true })
+    .withMessage("avatar can't be null or empty.")
+    .blacklist("<>")
+    .trim()
+    .isLength({ min: 1, max: 255 })
+    .withMessage("Name length must be between 1 to 255 characters.")
+    .isURL()
+    .withMessage("Must be an URL!")
+    .matches(/(?:\.jpg|\.png)$/)
+    .withMessage("Must be an image URL in .jpg or .png form!"),
+
+  // Age Check: be greater than or equal to 16.
+  body("age", "Invalid age.")
+    .exists({ checkFalsy: true })
+    .blacklist("<>")
+    .trim()
+    .toInt()
+    .custom((value) => {
+      if (value < 16) {
+        throw new Error("Age must be greater than 16.");
+      }
+      return true;
+    }),
+
+  // Interests Check: Must be an array.
+  body("interests", "Invalid interests.")
+    .blacklist("<>")
+    .trim()
+    .isArray()
+    .withMessage("Must be an array.")
+    .exists({ checkFalsy: true })
+    .withMessage("interests can't be null or empty."),
```

Comment:
We should be more strict and validate that this is:
- an array that only contains words (strings with only alphabet characters). You can use that with express-validator [wildcards](https://express-validator.github.io/docs/guides/field-selection/#wildcards)
- not an empty array by passing in length parameters in the `isArray()` check, see [here](https://express-validator.github.io/docs/api/validation-chain/#isarray)